contracts/governance/Pauser.sol

```diff
@@ -0,0 +1,316 @@
+     */
+    function initialize(
+        ITimelock _timelock,
+        IOwnedUpgradeabilityProxy _governor,
+        IVoteToken _trustToken,
+        IVoteToken _stkTRU,
+        uint256 _votingPeriod
+    ) external {
+        UpgradeableClaimable.initialize(msg.sender);
+        timelock = _timelock;
+        governor = _governor;
+        trustToken = _trustToken;
+        stkTRU = _stkTRU;
+        votingPeriod = _votingPeriod;
+    }
+
+    /**
+     * @dev Get the request state for the specified request
+     * @param requestId ID of a request in which to get its state
+     * @return Enumerated type of RequestState
+     */
+    function state(uint256 requestId) public view returns (RequestState) {
+        require(requestCount >= requestId && requestId > 0, "Pauser::state: invalid request id");
+        PauseRequest storage request = requests[requestId];
+        if (request.executed) {
+            return RequestState.Executed;
+        } else if (block.timestamp >= add256(request.eta, request.endTime)) {
+            return RequestState.Expired;
+        } else if (request.votes >= quorumVotes()) {
+            return RequestState.Succeeded;
+        } else if (block.timestamp <= request.endTime) {
+            return RequestState.Active;
+        } else {
+            return RequestState.Defeated;
+        }
+    }
+
+    /**
+     * @dev Create a request to pause the protocol or its parts
+     * @param targets The ordered list of target addresses for calls to be made during request execution
+     * @param methods The ordered list of function signatures to be passed during execution
+     * @return The ID of the newly created request
+     */
+    function makeRequest(address[] memory targets, PausingMethod[] memory methods) public returns (uint256) {
+        require(
+            countVotes(msg.sender, sub256(block.number, 1)) > requestThreshold(),
+            "Pauser::makeRequest: requester votes below request threshold"
+        );
+        require(targets.length == methods.length, "Pauser::makeRequest: request function information arity mismatch");
+        require(targets.length != 0, "Pauser::makeRequest: must provide actions");
+        require(targets.length <= requestMaxOperations(), "Pauser::makeRequest: too many actions");
+
+        uint256 latestRequestId = latestRequestIds[msg.sender];
+        if (latestRequestId != 0) {
+            RequestState proposersLatestRequestState = state(latestRequestId);
+            require(
+                proposersLatestRequestState != RequestState.Active,
+                "Pauser::makeRequest: one live request per proposer, found an already active request"
+            );
+        }
+
+        uint256 startBlock = block.number;
+        uint256 endTime = add256(block.timestamp, votingPeriod);
+
+        requestCount++;
+        PauseRequest memory newRequest = PauseRequest({
+            id: requestCount,
+            requester: msg.sender,
+            eta: EXECUTION_PERIOD,
```

Comment:
If every request is going to have the same "eta", then I would remove it from this struct and just use the constant `EXECUTION_PERIOD` in the place eta is currently used; it will make it easier to understand what's going on (and also save some gas on storage).  (also "eta" is an inconsistent name for this concept, since in GovernorAlpha it represented a timestamp while here it represents a time duration).